### ISSUE TITLE
Variable was defined twice when instantiating an array of structs in one expression

### ISSUE THREAD

**Issue by sirasistant** [noir-lang/noir#1925]
### Aim

The following should compile 
```rust
struct Point {
    x: Field,
    y: Field,
}

fn main()  {
    let arr = [Point { x: 0, y: 3 }];
}

```

### Expected Behavior

No errors!

### Bug

```
The application panicked (crashed).
Message:  Variable LocalId(0) was defined twice in ssa-gen pass
Location: crates/noirc_evaluator/src/ssa_refactor/ssa_gen/context.rs:400
```

### To Reproduce

1.
2.
3.
4.

### Installation Method

None

### Would you like to submit a PR for this Issue?

No

**Comment by jfecher** [comment]
Interesting, I've never seen this error before. I assume it is an issue with the array of structs to struct of arrays conversion that we want to disable soon anyway. I'll look into it to verify if that is the case and disable it sooner rather than later.

**Comment by vezenovm** [comment]
This error doesn't happen in https://github.com/noir-lang/noir/pull/1917 where I conditionally remove `aos_to_soa` checks for the experimental ssa.

**Comment by vezenovm** [comment]
Issue resolved following #1917 being merged into master
